src/Npgsql/NpgsqlDataSource.cs

```diff
@@ -247,6 +248,76 @@ public async Task ReloadTypesAsync(CancellationToken cancellationToken = default
         }
     }
 
+    /// <summary>
+    /// Attempts to return a mapping for a specific type.
+    /// </summary>
+    public PgTypeInfo? TryGetMapping<T>(string? dataTypeName = null) => TryGetMapping(typeof(T), dataTypeName);
+
+    /// <summary>
+    /// Attempts to return a mapping for a specific type.
+    /// </summary>
+    public PgTypeInfo? TryGetMapping(Type? type = null, string? dataTypeName = null)
```

Comment:
Note that PgTypeInfo is [Experimental], and also in the Internal namespace - I think that up to now the idea was that this was only for converter writers; we'd be exposing this to users for the first time. This also shows in the public API surface of this type - most of the stuff there probably shouldn't be shown to users.

I think that if we want to add an introspection API like this, we need another type. We do have the PostgresType hierarchy, but that really models the PG type only (no CLR type there, for example).

(just noting that there's no real urgent need for this API from my side)

/cc @NinoFloris

Reply:
> I think that if we want to add an introspection API like this, we need another type. We do have the PostgresType hierarchy, but that really models the PG type only (no CLR type there, for example).

Makes sense. We just have to agree on what exactly that type has to have. Probably, CLR type, the name of postgres's type + whether it supports writing?

Reply:
NpgsqlMappingInfo or something? Maybe @NinoFloris has a good idea here.

It could actually reference the PostgresType rather than just a string name (for full introspection capabilities). Besides that, just having the CLR type seems sufficient to me as a start, writability seems like an extra thing we don't necessarily need to do...